docs/DeveloperGuide.md

```diff
@@ -182,24 +186,85 @@ The above process is shown in the following sequence diagram:
 The following activity diagram summarizes what happens when a user executes a new command:
 ![DoneCommandActivityDiagram](images/DoneCommandActivityDiagram.png)
 
-### \[Proposed\] Data archiving
+### 2. Data archiving
+
+`[written by: Jin Hao]`
+
+The data archiving feature will archive all past appointments into an archive directory on starting the app.
+The appointment data will be archived according to their months and saved as a csv file.
+
+The data stored on the archive will be minimal and only contains the following columns: `date`, `startTime`, `endTime`, `isDone`, `name`, `phone`, `address` and `remark`.
+
+#### Implementation
+
+The archive mechanism is facilitated by `CsvAppointmentArchive` which implements the `AppointmentArchive` interface.
+It is stored internally within the `JsonAppointmentBookStorage` which in turn implements the `AppointmentBookStorage` interface.
+
+`CsvAppointmentArchive` implements the following operations:
+
+* `AppointmentArchive#archivePastAppointments(ReadOnlyAppointmentBook)` — Removes all past appointments from the `ReadOnlyAppointmentBook` and archive them as a csv file.
+* `AppointmentArchive#saveAppointments(List<CsvAdaptedAppointment>, String)` — Saves the list of `CsvAdaptedAppointment` as a csv file in the archive directory with the given filename.
+* `AppointmentArchive#readAppointments(String)` — Reads the csv file with the given filename and returns the data as a `List<CsvAdaptedAppointment>`
+* `AppointmentArchive#getArchiveStatistics()` — Gets the status message of the archive mechanism.
+
+Of these three, only the `archivePastAppointments(ReadOnlyAppointmenBook)` and `getArchiveStatistics()` are exposed in the `AppointmentBookStorage` and `Storage` interfaces as methods with the same signature.
+
+`CsvAdaptedAppointment` and `CsvAdaptedPatient` are used to represent the csv-adapted `Appointment` and `Patient` respectively.
+
+Given below is an example archive run scenario and how the archive mechanism behaves at each step.
+
+1. The user launches the application with some existing appointment data (not launching for the first time).
```

Comment:
Sorry that I'm a bit confused why all the numbering is `1`?
Is it supposed to be that?

Reply:
`1.` is actually a markdown format for numbering bullet point. They will automatically assign the number to it, this way don't have to manually update the numbering everytime a point is added or removed.

Can take a look at this for the markdown preview: [Dev Guide](https://github.com/JinHao-L/tp/blob/dev-guide-archive/docs/DeveloperGuide.md)